clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CommitApplicationEvent.java

```diff
@@ -19,22 +19,23 @@
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
 
+import java.util.Collections;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 
-public class CommitApplicationEvent extends ApplicationEvent {
-    final private CompletableFuture<Void> future;
+public class CommitApplicationEvent extends CompletableApplicationEvent<Void> {
     final private Map<TopicPartition, OffsetAndMetadata> offsets;
 
     public CommitApplicationEvent(final Map<TopicPartition, OffsetAndMetadata> offsets) {
         super(Type.COMMIT);
-        this.offsets = offsets;
-        Optional<Exception> exception = isValid(offsets);
-        if (exception.isPresent()) {
-            throw new RuntimeException(exception.get());
+        this.offsets = Collections.unmodifiableMap(offsets);
+
+        for (OffsetAndMetadata offsetAndMetadata : offsets.values()) {
```

Comment:
This is an existing issue. Could we change `toString` to include the fields in the parent class? Do we need to add `equals` and `hashcode` for consistency?

Reply:
There are quite a few events out there that need to be refactored if we do it this way.  Would it be ok for you to skip this for now and I will post a MINOR patch to address after this is closed?

Reply:
If your PR goes in before mine, you can make a comment and I'll add the `toString` changes into mine.

Reply:
Yes, it's fine to clean this up separately.

Reply:
@junrao - per your request.  Here's the patch to address your comment: https://github.com/apache/kafka/pull/14391